### PR TITLE
Expand touch area for Android selection handles (Resolves #2075)

### LINKED ISSUE #2075: [SuperEdiror][Android] Selection Handle touch area is too small
**Package Version**
stable

See the difference of touch area between the native selection handle and the SuperEditor ones.
**Native message app, Android simulator:**

https://github.com/superlistapp/super_editor/assets/6229343/7be996bb-553f-4a47-bedf-f71010501770

**SuperEditor demo app, Android Simulator:**

https://github.com/superlistapp/super_editor/assets/6229343/9330bdac-dd49-4849-9f24-85240101cc5d

I heard users complaining about selection and I think this issue can explain some of their frustration.

### PR DESCRIPTION
**Pull request by matthew-carroll**
Expand touch area for Android selection handles (Resolves #2075)

I added an invisible circle around each selection handle to expand the touch area. This added invisible padding also required an adjustment to the offset applied to the handles as followers of the selection bounds widgets.

I think that our `Follower` widget doesn't apply the device pixel ratio to its internal math, but I think that maybe `CompositedTransformFollower` does, which is why our `Follower` offsets also incorporate the device pixel ratio.

### CODE REVIEW

**Comment by matthew-carroll** [comment]
@BazinC please try out this PR and let us know if it solves your selection issues.

File: super_editor/test_goldens/editor/mobile/goldens/mobile-selection_android_drag-base-upstream.png
Comment (angelosilvestre) [inline super_editor/test_goldens/editor/mobile/goldens/mobile-selection_android_drag-base-upstream.png:1]: In this golden, and on others, the red line changed to close to the bottom of the handle. Is this expected? Will this impact anything?

File: super_editor/test_goldens/editor/mobile/mobile_selection_test.dart [hunk 1]
```diff
@@ -115,10 +116,20 @@ void main() {
...
-            await tester.dragFrom(handleRectGlobal.center, dragDelta);
+
+            // Calculate the center of the visual handle, accounting for addition of invisible
+            // touch area expansion. The touch area only expands below the handle, not above
+            // the handle, so using the "center" of the widget would product a point that's
+            // too far down. The invisible height below the handle is about the same height as
+            // the handle, so we'll use the 25% y-value of the whole widget, which is roughly
+            // at the 50% y-value of the visible handle.
+            final centerOfVisualHandle =
+                Offset(handleRectGlobal.center.dx, handleRectGlobal.top + handleRectGlobal.height / 4);
```
Comment (angelosilvestre) [inline super_editor/test_goldens/editor/mobile/mobile_selection_test.dart:127]: After reading this I think I understand why the goldens changed.

**Review by angelosilvestre — APPROVED** [review]
LGTM

**Comment by BazinC** [comment]
Hey @matthew-carroll I tried your PR. It expands the touch area as shown in red in the screenshot below.

<p align="center">
<\img>
</p>

But compared to native Android editor, it feels like the native one is more permissive.
I tried to find the correct "expanded touch area" in android code source (here [HandleView](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/widget/Editor.java;bpv=1;bpt=1;l=4937?q=HandleView&ss=android%2Fplatform%2Fsuperproject%2Fmain&gsn=HandleView&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fandroid.googlesource.com%2Fplatform%2Fsuperproject%2Fmain%2F%2Fmain%3Flang%3Djava%3Fpath%3Dandroid.widget.Editor.HandleView%233607f72254be2208c5e6bf477c8037f616e185e8c933cc5872481108c8ed081a)) but it seems the logic would require more than a sized GestureDetector to replicate the native feel. (see mIdealVerticalOffset, mIdealFingerToCursorOffset...)

How about merging this PR as a quick fix to the "too small touch area" problem and revisit it with a new ticket about cloning native behavior?

**Comment by matthew-carroll** [comment]
@BazinC can you please file that new ticket with the details that will be needed to implement it? We'll need you to be a lot more specific than just "permissive" or other high level descriptions. We'll need to capture the desired heuristic behaviors.

**Pull request merged**